src/tools/rbd_mirror/InstanceWatcher.cc

```diff
@@ -65,6 +68,28 @@ struct RemoveInstanceRequest : public Context {
   }
 };
 
+struct NotifyInstanceRequest : public Context {
+  librbd::watcher::Notifier notifier;
+  bufferlist bl;
+  AsyncOpTracker &op_tracker;
+
+  NotifyInstanceRequest(librados::IoCtx &io_ctx, ContextWQ *work_queue,
+                        const std::string &instance_id, bufferlist &&bl,
+			AsyncOpTracker &op_tracker)
+    : notifier(work_queue, io_ctx, RBD_MIRROR_INSTANCE_PREFIX + instance_id),
+      bl(bl), op_tracker(op_tracker) {
+    op_tracker.start_op();
+  }
+
+  void send() {
+    notifier.notify(bl, nullptr, this);
```

Comment:
Might be nice to update the API to ```Notifier::notify``` to replace the ```out_bl``` with a pointer to a response struct (map from <gid, cookie> to response payload and list of timed out <gid, cookie> peers). If the response struct has zero peers in the response, you know no instances were registered on the watch.